src/imitation/util/logger.py

```diff
@@ -0,0 +1,126 @@
+  raise RuntimeError("Shouldn't call stable_baselines.logger.configure "
+                     "once imitation.logger.configure() has been called")
+
+
+def _sb_logger_reset_replacement():
+  raise NotImplementedError("reset() while using imitation.configure()")
+
+
+_in_accumul_context = False
+_configured = False
+_format_strs = None
+
+
+def configure(folder: str, format_strs: Optional[Sequence[str]] = None) -> None:
+  """Configure Stable Baselines logger to be `accumulate_means()`-compatible.
+
+  After this function is called, `stable_baselines.logger.{configure,reset}()`
+  are replaced with stubs that raise errors (not yet implemented).
+
+  Args:
+      folder: Argument from `stable_baselines.logger.configure`.
+      format_strs: Argument from `stable_baselines.logger.configure`.
+  """
+  global _configured, _format_strs
+  assert not _in_accumul_context
+  _configured = True
+  _format_strs = format_strs
+
+  sb_logger.configure = _sb_logger_configure_replacement
+  sb_logger.reset = _sb_logger_reset_replacement
+
+  output_formats = _build_output_formats(folder, format_strs)
+  sb_logger.Logger.DEFAULT = sb_logger.Logger(folder, output_formats)
+  sb_logger.Logger.CURRENT = sb_logger.Logger.DEFAULT
+  sb_logger.log('Logging to %s' % folder)
+
+
+def dumpkvs():
+  """Alias for `stable_baselines.logger.logkv`."""
+  sb_logger.dumpkvs()
+
+
+@contextlib.contextmanager
+def accumulate_means(subdir_name: str):
+  """Temporarily redirect logkv() to a different logger and auto-track kvmeans.
+
+  Within this context, the original logger is swapped out for a special logger
+  in directory `"{current_logging_dir}/raw/{subdir_name}"`.
+
+  The special logger's `stable_baselines.logger.logkv(key, val)`, in addition
+  to tracking its own logs, also forwards the log to the original logger's
+  `.logkv_mean()` under the key `mean/{subdir_name}/{key}`.
+
+  After the context exits, these means can be dumped as usual using
+  `stable_baselines.logger.dumpkvs()` or `imitation.util.logger.dumpkvs()`.
+
+  This context cannot be nested.
+
+  Args:
+    subdir_name: Chooses the logger subdirectories and temporary logger.
+  """
+  global _in_accumul_context
+  assert _configured
+  assert not _in_accumul_context
+  _in_accumul_context = True
+
+  try:
+    sb_logger.Logger.CURRENT = _AccumulatingLogger.from_subdir(subdir_name)
+    yield
+  finally:
+    # Switch back to default logger.
+    sb_logger.Logger.CURRENT = sb_logger.Logger.DEFAULT
+    _in_accumul_context = False
+
+
+class _AccumulatingLogger(sb_logger.Logger):
+
+  _cached_loggers: Dict[str, "_AccumulatingLogger"] = {}
+
+  def __init__(self,
+               folder,
+               output_formats,
+               *,
+               subdir: str):
+    """Like Logger, except also accumulates logkv_mean on default logger."""
+    super().__init__(folder, output_formats)
+    self.subdir = subdir
+
+  def logkv(self, key, val):
+    raw_key = os.path.join("raw", self.subdir, key)
```

Comment:
@qxcv In case this is helpful for quickly understanding which prefixes are used where, `test_logger.py::test_hard` shows the logger in action

Reply:
The motivation is that you need different prefixes for averaged stuff vs. not averaged stuff, right? If so the current arrangement seems fine to me: un-averaged data ends up in `raw/`, and averaged stats end up in `mean/`.

(It's really unfortunate that we can't have something like a relational database, where we just write everything to a file with appropriate foreign keys & then do all our averaging during post-processing. I still can't think of a way to get that with CSV, though, unless we're willing to limit ourselves to only using Pandas for analysis & throw away support for stuff like viskit.)

Reply:
My point was that this particular logger *only* has keys `raw/{self.subdir}`, with the master logger having the `mean/...` keys. Are you imagining concatenating all the CSVs together? I'm not sure that'll work however we set it up.

Reply:
@AdamGleave Why would we need to concatenate data from the raw CSVs together? Isn't the main point of having the means logging so that we can sidestep comparing logs with conflicting timesteps?